Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -530,10 +530,38 @@ std::optional<int64_t> parseUTCOffsetInMinutes(StringView string)
 template<typename CharacterType>
 static bool canBeCalendar(const StringParsingBuffer<CharacterType>& buffer)
 {
-    // https://tc39.es/proposal-temporal/#prod-Calendar
-    // Calendar :
-    //     [u-ca= CalendarName]
-    return buffer.lengthRemaining() >= 6 && buffer[0] == '[' && buffer[1] == 'u' && buffer[2] == '-' && buffer[3] == 'c' && buffer[4] == 'a' && buffer[5] == '=';
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    // Step 4(a)(ii)(2)(a):
+    //  Let key be the source text matched by the AnnotationKey Parse Node contained within annotation
+    // This just checks for '[', followed by an optional '!' (critical flag),
+    // followed by a valid key, followed by an '='.
+
+    int32_t len = buffer.lengthRemaining();
+    // Check for [! or [, followed by any number of 'a'-'z' or '-', followed by '='
```

Comment:
Why is it `int32_t`? This looks not correct since it should be `size_t`.
And we should early return here when the length is 0.
Also, let's use `length` instead of `len`.

Reply:
Fixed in [19227db](https://github.com/WebKit/WebKit/pull/36361/commits/19227db637e5c5b34d4676a07f7a0147f9f991fa)